code/extensions/che-api/src/impl/github-service-impl.ts

```diff
@@ -16,40 +16,50 @@ import { AxiosInstance } from 'axios';
 import * as fs from 'fs-extra';
 import * as path from 'path';
 
+const GITHUB_GET_AUTHENTICATED_USER = 'https://api.github.com/user';
+
 @injectable()
 export class GithubServiceImpl implements GithubService {
     private readonly token: string | undefined;
 
     constructor(@inject(Symbol.for('AxiosInstance')) private readonly axiosInstance: AxiosInstance) {
         const credentialsPath = path.resolve('/.git-credentials', 'credentials');
         if (fs.existsSync(credentialsPath)) {
-            const token = fs.readFileSync(credentialsPath).toString();
-            this.token = token.substring(token.lastIndexOf(':') + 1, token.indexOf('@'));
+            const content = fs.readFileSync(credentialsPath).toString();
+
+            // Since this service is specific for github.com, we need to find only one corresponding token
+            const regex = /(http|https):\/\/oauth2:(\d|\S+)@github.com$/mg;
```

Comment:
we should test it for the case when custom git server is used
I mean - before your changes a token was read this way:
```
this.token = token.substring(token.lastIndexOf(':') + 1, token.indexOf('@'));
```
As far as I understand, now it depends on the `github.com` pattern, so it can bring a regression for the custom git server

Reply:
I suppose we are talking about GitHub, which is `github.com`

Reply:
Anyway, I was surprised it works with token 'anykey'.